src/python/pants/backend/shell/goals/test.py

```diff
@@ -92,6 +91,12 @@ async def test_shell_command(
         ),
     )
 
+    if field_set.cache_scope.value is None and test_subsystem.force:
+        shell_process = dataclasses.replace(
+            shell_process,
+            cache_scope=ProcessCacheScope.PER_SESSION,
```

Comment:
Im assuming the default is "SUCCESSFUL" then?

Reply:
The default is "successful" for all environments other than the workspace environment. The issue was that this code was overwriting the "per session" default of the workspace environment with "successful."